src/btree/bt_read.c

```diff
@@ -321,6 +323,38 @@ __las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
             WT_ERR(
               cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
 
+            /*
+             * If our update is a modify then squash it into a standard update. It's a problem if we
+             * need to read backwards into lookaside just to make sense of what we have in our
+             * update list.
+             */
+            if (upd_type == WT_UPDATE_MODIFY) {
+                while (upd_type == WT_UPDATE_MODIFY) {
+                    WT_ERR(__wt_update_alloc(session, &las_value, &upd, &incr, upd_type));
+                    if (mod_counter >= WT_MODIFY_ARRAY_SIZE) {
+                        if (mod_counter == WT_MODIFY_ARRAY_SIZE)
+                            listp = NULL;
+                        WT_ERR(
+                          __wt_realloc_def(session, &allocated_bytes, mod_counter + 1, &listp));
+                        if (mod_counter == WT_MODIFY_ARRAY_SIZE)
+                            memcpy(listp, list, sizeof(list));
+                    }
+                    listp[mod_counter++] = upd;
+                    WT_ERR(cursor->prev(cursor));
+                    WT_ERR(cursor->get_value(
+                      cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+                }
+                WT_ASSERT(session, upd_type == WT_UPDATE_STANDARD);
```

Comment:
Ahh yes, what would happen in the case that the base update is the on-disk value?

Similarly, what would happen if the value searched by `__wt_find_lookaside_upd()` points to a modify record, but the base for that modify happens to be the on-disk value?

Reply:
I think this is fine because this code path is explicitly when there is no birthmark. If there is, then we will use that to create a birthmark record (which can't be a modify).

However, I agree about `__wt_find_lookaside_upd`. This is actually causing an assertion failure in @kommiharibabu's PR so I'm looking into that now.

Reply:
cool, I agree.

Reply:
got it.